frontend/src/AssignmentView/AddAssignmentModal.tsx

```diff
@@ -1,35 +1,76 @@
-import React, { useState } from "react";
+import React, { FormEvent, useContext, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Container from "react-bootstrap/Container";
 import Form from "react-bootstrap/Form";
 import Button from "react-bootstrap/Button";
 import Alert from "react-bootstrap/Alert";
 import Stack from "react-bootstrap/Stack";
 import Modal from "react-bootstrap/Modal";
+import { userContext } from "../Context/UserContext";
+import { ProfessorAssignmentContext } from "../Context/ProfessorAssignmentContext";
+import { useCookies } from "react-cookie";
 
 // This view is a form for professors to add a assignment
 function AddAssignmentModal() {
+  const user = useContext(userContext);
   const [error, setError] = useState("");
+  const [cookies, setCookies] = useCookies(["jwt"]);
   const nav = useNavigate();
+  const classId = 1;
+  const assignmentId = 1;
 
   const [show, setShow] = useState(false);
+  const [gradingScriptName, setGradingScriptName] = useState("");
+  const [gradingScriptFile, setGradingScriptFile] = useState("");
+  const [dueDate, setDueDate] = useState("");
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
 
-  const submit: React.FormEventHandler<HTMLFormElement> = (
-    e: React.FormEvent<HTMLFormElement>
-  ) => {
+  function handleGradingScript(data: FormData) {
+    return fetch(
+      `http://localhost:8080/class/${classId}/${assignmentId}/upload`,
+      {
+        method: "POST",
+        mode: "cors",
+        headers: {
+          Authorization: cookies.jwt,
+        },
+        body: data,
+      }
+    )
+      .then((response) => {
+        if (response.status == 401) throw "Unauthorized";
+        return response.json();
+      })
+      .catch((e) => {
+        setError(
+          `Failed uploading! Server responded with: ${String(e).replace(
+            "TypeError: ",
+            ""
+          )}`
+        );
+      });
+  }
+
+  async function submit(e: FormEvent<HTMLFormElement>) {
     e.preventDefault();
-    // send assignment data to database
-    // set up autograding (?)
-    setError((err) => (err ? "" : `File type is not supported`));
-    nav("/professor/class/");
-  };
+    console.log("hey");
+    const data = new FormData(e.currentTarget);
+    handleGradingScript(data);
+    // TODO: The API call to submit a file is still unimplemented. Will fix when implemented
+    //setError((err) => (err ? "" : `File type is not supported`));
+    //nav("/professor/class/");
```

Comment:
Remove. API call is implemented. Rebase on main and uncomment this `nav()` call.